src/flow/assembly_lmh.hh

```diff
@@ -235,139 +232,144 @@ protected:
-                        double bc_total_flux = bc_flux + bc_sigma*(bc_switch_pressure - bc_pressure);
-                        
-                        loc_system_.add_value(edge_row, bc_total_flux * b_ele.measure() * cross_section);
-                    }
-                } 
-                else {
-                    xprintf(UsrErr, "BC type not supported.\n");
+            } else {
+                // check and possibly switch to  pressure BC
+                // TODO: What is the appropriate DOF in not local?
+                // The switch raise error on the corresponding side row.
+                // Magnitude of the error is abs(solution_head - bc_pressure)
+                // Since usually K is very large, this error would be much
+                // higher then error caused by the inverse switch, this
+                // cause that a solution  with the flux violating the
+                // flux inequality leading may be accepted, while the error
+                // in pressure inequality is always satisfied.
+
+                double solution_head = ad_->p_edge_solution[loc_schur_.row_dofs[sidx]];
+
+                if ( solution_head > bc_pressure) {
+                    //DebugOut().fmt("x: {}, to dirich, p: {} -> p: {} f: {}\n",b_ele.centre()[0], solution_head, bc_pressure, bc_flux);
+                    switch_dirichlet=1;
                 }
             }
-            loc_system_.add_value(side_row, edge_row, 1.0);
-            loc_system_.add_value(edge_row, side_row, 1.0);
+            }
+
+            save_local_system_ = (bool) switch_dirichlet;
+            
+            // ** Apply BCUpdate BC type. **
+            // Force Dirichlet type during the first iteration of the unsteady case.
+            if (switch_dirichlet || ad_->force_bc_switch ) {
+                //DebugOut().fmt("x: {}, dirich, bcp: {}\n", b_ele.centre()[0], bc_pressure);
+                loc_schur_.set_solution(sidx, bc_pressure);
+                dirichlet_edge[sidx] = 1;
+            } else {
+                //DebugOut()("x: {}, neuman, q: {}  bcq: {}\n", b_ele.centre()[0], side_flux, bc_flux);
+                loc_system_.add_value(edge_row, side_flux);
+            }
+
+        } else if (type==DarcyMH::EqData::river) {
+            ad_->is_linear=false;
+
+            double bc_pressure = ad_->bc_pressure.value(b_ele.centre(), b_ele);
+            double bc_switch_pressure = ad_->bc_switch_pressure.value(b_ele.centre(), b_ele);
+            double bc_flux = -ad_->bc_flux.value(b_ele.centre(), b_ele);
+            double bc_sigma = ad_->bc_robin_sigma.value(b_ele.centre(), b_ele);
+
+            double solution_head = ad_->p_edge_solution[loc_schur_.row_dofs[sidx]];
+
+            // Force Robin type during the first iteration of the unsteady case.
+            if (solution_head > bc_switch_pressure  || ad_->force_bc_switch) {
```

Comment:
'force_bc_switch' rename to 'force_no_neuman_bc' ?

Reply:
Agreed.